### PR TITLE
1060:vpd-tool:Check for read only FRUs before write

### PR DESCRIPTION
**Pull request by PriyangaRamasamy**
This commit fixes the issue where vpd-tool tries to perform write on request for the FRUs which are read only.

Test:
./vpd-tool -w -O /system/chassis/motherboard/vrm0 -R VINI -K CC --value 51F3
Read only FRU. Update not allowed.

### CODE REVIEW

File: config/ibm/50003000.json [hunk 1]
```diff
@@ -571,6 +571,7 @@
         "/sys/bus/i2c/drivers/at24/9-0052/eeprom": [
             {
                 "inventoryPath": "/system/chassis/motherboard/vrm8",
+                "readOnlyFRU": 1,
```
Comment (jinuthomas) [inline config/ibm/50003000.json:574]: Why define it as FRU when implicit

File: vpd_tool.cpp [hunk 2]
```diff
@@ -128,6 +128,11 @@ int main(int argc, char** argv)
 
         if (*writeFlag)
         {
+            if (isReadOnlyFRU(objectPath, jsObject) == 1)
+            {
+                throw runtime_error("Read only FRU. Update not allowed.");
```
Comment (jinuthomas) [inline vpd_tool.cpp:133]: nit FRU --> EEPROM

File: ibm_vpd_utils.hpp [hunk 3]
```diff
@@ -567,5 +567,15 @@ void findBackupVPDPaths(std::string& backupEepromPath,
...
+/**
+ * @brief Check if given FRU is read only or not.
+ *
+ * @param[in] vpdPath - VPD path of the FRU.
+ * @param[in] jsObject - Inventory JSON object.
+ *
+ * @return 0 if FRU is not read only. 1 if FRU is read only. -1 on processing
+ * failures like invalid JSON/empty VPD path/path not found in JSON.
+ */
+int isReadOnlyFRU(const std::string& vpdPath, const nlohmann::json& jsObject);
```
Comment (jinuthomas) [inline ibm_vpd_utils.hpp:579]: better : isWritableEeprom

File: ibm_vpd_utils.cpp [hunk 4]
```diff
@@ -1221,5 +1221,47 @@ void getBackupRecordKeyword(std::string& record, std::string& keyword)
         }
     }
 }
+
+int isReadOnlyFRU(const std::string& vpdPath, const nlohmann::json& jsObject)
+{
+    if (vpdPath.empty())
```
Comment (jinuthomas) [inline ibm_vpd_utils.cpp:1227]: This should be common in the main part of the tool not here

File: ibm_vpd_utils.hpp [hunk 5]
```diff
@@ -567,5 +567,15 @@ void findBackupVPDPaths(std::string& backupEepromPath,
  */
 void getBackupRecordKeyword(std::string& record, std::string& keyword);
 
+/**
+ * @brief Check if given FRU is read only or not.
+ *
+ * @param[in] vpdPath - VPD path of the FRU.
+ * @param[in] jsObject - Inventory JSON object.
+ *
+ * @return 0 if FRU is not read only. 1 if FRU is read only. -1 on processing
```
Comment (jinuthomas) [inline ibm_vpd_utils.hpp:576]: this should be boolean , any API which is a question should return only. true or false.

**Review by jinuthomas — CHANGES_REQUESTED** [review]

File: config/ibm/50003000.json (hunk 1 above)
Comment (PriyangaRamasamy) [reply config/ibm/50003000.json]: modified

File: ibm_vpd_utils.cpp (hunk 4 above)
Comment (PriyangaRamasamy) [reply ibm_vpd_utils.cpp]: done

File: ibm_vpd_utils.hpp (hunk 3 above)
Comment (PriyangaRamasamy) [reply ibm_vpd_utils.hpp]: may I know why it is better?

File: vpd_tool.cpp (hunk 2 above)
Comment (PriyangaRamasamy) [reply vpd_tool.cpp]: done

File: ibm_vpd_utils.hpp (hunk 5 above)
Comment (PriyangaRamasamy) [reply ibm_vpd_utils.hpp]: done

**Review by SunnySrivastava1984 — APPROVED** [review]

**Pull request merged**